File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.java
class Solution {
    public boolean checkOnesSegment(String s) {
        int n = s.length();
        boolean oneFound = false;

        for (int i = 0; i < n; i++) {
            if (s.charAt(i) == '1' && !oneFound) {
                oneFound = true;
            }
            if (oneFound && s.charAt(i) == '1') {
                if (i >= 1 && s.charAt(i - 1) == '0') {
                    return false;
                }
            }
        }
        return true;
    }
}

File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        n = len(s)
        oneFound = False

        for i in range(n):
            if(s[i] == '1' and not oneFound):
                oneFound = True
            if(oneFound and s[i] == '1'):
                if( i >= 1 and s[i-1] == '0'): return False
        return True